cluster_experiments/experiment_analysis.py

```diff
@@ -301,6 +301,92 @@ def from_config(cls, config):
         )
 
 
+class PairedTTestClusteredAnalysis(ExperimentAnalysis):
+    """
+    Class to run paired T-test analysis on aggregated data
+
+    Arguments:
+        cluster_cols: list of columns to use as clusters
+        target_col: name of the column containing the variable to measure
+        treatment_col: name of the column containing the treatment variable
+        treatment: name of the treatment to use as the treated group
+        strata: column to generate 2 groups (a and b)
+
+    Usage:
+
+    ```python
+    from cluster_experiments.experiment_analysis import TTestClusteredAnalysis
+    import pandas as pd
+
+    df = pd.DataFrame({
+        'x': [1, 2, 3, 4, 0, 0, 1, 1],
+        'treatment': ["A", "B", "A", "B"] * 2,
+        'cluster': [1, 2, 3, 4, 1, 2, 3, 4],
+    })
+
+    PairedTTestClusteredAnalysis(
+        strata=df['cluster'],
+        target_col=df['x'],
+    ).get_pvalue(df)
+    ```
+    """
+
+    def __init__(
+        self,
+        cluster_cols: List[str],
+        strata_cols: List[str] = None,
```

Comment:
doesn't follow type hint, if you want it to be None, should be Optional[List[str]]. I would go for not allowing it to be none anyway

Reply:
So is it preferable to have a mandatory strata_cols?

Reply:
I think so, the analysis method will fail otherwise, right?